tests/annotations/tests.py

```diff
@@ -148,6 +148,19 @@ def test_mixed_type_annotation_numbers(self):
         combined = int(test.pages + test.rating)
         self.assertEqual(b.combined, combined)
 
+    def test_empty_expression_annotation(self):
+        books = Book.objects.annotate(
+            selected=ExpressionWrapper(Q(pk__in=[]), output_field=BooleanField())
```

Comment:
Yes, I agree. Leave the test and the compiler changes in. I'll create a new ticket for full Q expression support.